### PR TITLE
Feature invite friend implementation

### PR DESCRIPTION
**Pull request by piya43044**
Here's the pull request Description format:
- https://expense-management-system-batch-1-2022.atlassian.net/browse/EMSB12-83

**Checklist**
- [x] Code adheres to **[Git and Pull Request Guidelines]**
- [x] Code adheres to **[Angular Guidelines]**
- [x] Code adheres to **[Front-end Checklist]**

### CODE REVIEW

File: src/app/friends/friends-add-edit/friends-add-edit.component.ts [hunk 1]
```diff
@@ -45,9 +55,14 @@ export class FriendsAddEditComponent implements OnInit {
     return this.addFriendForm.get('friendName');
   }
 
-  // Submit method
+  // Submit method for invite friend
```
Comment (AshaPatel216) [inline src/app/friends/friends-add-edit/friends-add-edit.component.ts]: method comment format

File: src/app/friends/friends-add-edit/friends-add-edit.component.ts [hunk 2]
```diff
@@ -45,9 +55,14 @@ export class FriendsAddEditComponent implements OnInit {
...
   onSubmit(): void{
-    this.addFriendForm.reset();
-    this.router.navigate(['friends/friends-list']);
+    this.loading = true;
+    this.friendService.postFriend(this.addFriendForm.value).subscribe( response => {
```
Comment (AshaPatel216) [inline src/app/friends/friends-add-edit/friends-add-edit.component.ts]: what id API throws an error> handle errors too

Comment (AshaPatel216) [reply src/app/friends/friends-add-edit/friends-add-edit.component.ts]: example:
` public getRepos() {
    this.loading = true;
    this.errorMessage = "";
    this.githubService.getReposCatchError(this.userName)
      .subscribe(
        (response) => {                           //Next callback
          console.log('response received')
          this.repos = response;
        },
        (error) => {                              //Error callback
          console.error('error caught in component')
          this.errorMessage = error;
          this.loading = false;
    
          //throw error;   //You can also throw the error to a global error handler
        }
      )
  }`

Comment (AshaPatel216) [reply src/app/friends/friends-add-edit/friends-add-edit.component.ts]: DO not use console log. instead use bootstrap toaster to show success and error message

File: src/app/services/friend.service.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+export class FriendService {
+
+  private baseUrl = "https://localhost:44329/api/app/add-friend/friend"
+
+  constructor(private http: HttpClient) { }
+
+  /**
+   * Post the data of friend in the server
+   * @param friend - Friend
+   * @returns Observable<Friend>
```
Comment (AshaPatel216) [inline src/app/services/friend.service.ts]: Add description of params and the returns
https://marketplace.visualstudio.com/items?itemName=salbert.comment-ts

File: src/app/friends/friends-add-edit/friends-add-edit.component.ts (hunk 1 above)
Comment (piya43044) [reply src/app/friends/friends-add-edit/friends-add-edit.component.ts]: done

File: src/app/friends/friends-add-edit/friends-add-edit.component.ts (hunk 2 above)
Comment (piya43044) [reply src/app/friends/friends-add-edit/friends-add-edit.component.ts]: di toaster is not working so I am using alert box.

File: src/app/services/friend.service.ts (hunk 3 above)
Comment (piya43044) [reply src/app/services/friend.service.ts]: done

File: src/app/friends/friends-add-edit/friends-add-edit.component.ts [hunk 4]
```diff
@@ -36,18 +47,32 @@ export class FriendsAddEditComponent implements OnInit {
...
-  // Submit method
+  /**
+   * Submit method for invite friend
+   *  */ 
   onSubmit(): void{
-    this.addFriendForm.reset();
-    this.router.navigate(['friends/friends-list']);
+    this.loading = true;
+    this.friendService.postFriend(this.addFriendForm.value).subscribe( (response) => {
+      // this.loading = false;
```
Comment (AshaPatel216) [inline src/app/friends/friends-add-edit/friends-add-edit.component.ts]: remove commented code

File: src/app/friends/friends-add-edit/friends-add-edit.component.ts (hunk 2 above)
Comment (AshaPatel216) [reply src/app/friends/friends-add-edit/friends-add-edit.component.ts]: Okay for now but please check why toaster is not working. DM me the problem

**Comment by AshaPatel216** [comment]
You can use https://www.npmjs.com/package/ngx-toastr for toaster in future. That is fine for alert for now

File: src/app/friends/friends-add-edit/friends-add-edit.component.ts (hunk 4 above)
Comment (piya43044) [reply src/app/friends/friends-add-edit/friends-add-edit.component.ts]: done

**Review by AshaPatel216 — APPROVED** [review]

**Pull request merged**